src/ApparentHorizons/ComputeItems.hpp

```diff
@@ -33,23 +35,29 @@ namespace Tags {
 template <size_t Dim, typename Frame>
 struct InverseSpatialMetricCompute : gr::Tags::InverseSpatialMetric<Dim, Frame>,
                                      db::ComputeTag {
-  static tnsr::II<DataVector, Dim, Frame> function(
+  using return_type = tnsr::II<DataVector, Dim, Frame>;
+  static void function(
+      const gsl::not_null<tnsr::II<DataVector, Dim, Frame>*> result,
       const tnsr::aa<DataVector, Dim, Frame>& psi) noexcept {
-    return determinant_and_inverse(gr::spatial_metric(psi)).second;
+    destructive_resize_components(result, psi.begin()->size());
+    *result = determinant_and_inverse(gr::spatial_metric(psi)).second;
   };
   using argument_tags = tmpl::list<gr::Tags::SpacetimeMetric<Dim, Frame>>;
   using base = gr::Tags::InverseSpatialMetric<Dim, Frame>;
 };
 template <size_t Dim, typename Frame>
 struct ExtrinsicCurvatureCompute : gr::Tags::ExtrinsicCurvature<Dim, Frame>,
                                    db::ComputeTag {
-  static tnsr::ii<DataVector, Dim, Frame> function(
+  using return_type = tnsr::ii<DataVector, Dim, Frame>;
+  static void function(
+      const gsl::not_null<tnsr::ii<DataVector, Dim, Frame>*> result,
       const tnsr::aa<DataVector, Dim, Frame>& psi,
       const tnsr::aa<DataVector, Dim, Frame>& pi,
       const tnsr::iaa<DataVector, Dim, Frame>& phi,
       const tnsr::II<DataVector, Dim, Frame>& inv_g) noexcept {
     const auto shift = gr::shift(psi, inv_g);
-    return GeneralizedHarmonic::extrinsic_curvature(
+    destructive_resize_components(result, psi.begin()->size());
+    *result = GeneralizedHarmonic::extrinsic_curvature(
```

Comment:
you should be able to call the mutating version of `GeneralizedHarmonic::extrinsic_curvature`